src/App.jsx

```diff
@@ -4,66 +4,140 @@ import TodoList from './components/TodoList';
-    };
+  const API_URL = `https://api.airtable.com/v0/${import.meta.env.VITE_AIRTABLE_BASE_ID}/${import.meta.env.VITE_TABLE_NAME}`;
+  const headers = {
+    Authorization: `Bearer ${import.meta.env.VITE_AIRTABLE_API_TOKEN}`,
+    'Content-Type': 'application/json'
+  };
 
-    const url = `https://api.airtable.com/v0/${import.meta.env.VITE_AIRTABLE_BASE_ID}/${import.meta.env.VITE_TABLE_NAME}`
+  const fetchData = async () => {
+    setIsLoading(true);
+    setError(null);
 
     try {
-      const response = await fetch(url, options);
+      const response = await fetch(`${API_URL}?view=Grid%20view&sort[0][field]=title&sort[0][direction]=${sortOrder}`, {
+        method: 'GET',
+        headers
+      });
 
       if (!response.ok) {
         throw new Error(`Error: ${response.status}`);
       }
       
       const data = await response.json();
 
+      // data.records.sort((objectA, objectB) => {
+      //   const titleA = objectA.fields.title.toLowerCase();
+      //   const titleB = objectB.fields.title.toLowerCase();
+
+      //   if (titleA < titleB) return -1;
+      //   if (titleA > titleB) return 1;
+      //   return 0
+      // });
+
+      // data.records.sort((objectA, objectB) => {
+      //   const titleA = objectA.fields.title.toLowerCase();
+      //   const titleB = objectB.fields.title.toLowerCase();
+
+      //   if (titleA < titleB) return 1;
+      //   if (titleA > titleB) return -1;
+      //   return 0
+      // });
+
       const todos = data.records.map((todo) => ({
           title: todo.fields.title,
-          id: todo.id
+          id: todo.id,
+          createdTime: todo.fields.createdTime || new Date().toISOString()
       }));
 
-    setTodoList(todos);
-    setIsLoading(false);
+      setTodoList(todos);
 
     } catch (error) {
-      console.log(`Erro fetching data: ${error.message}`);
+      setError(error.message)
+    } finally {
+      setIsLoading(false)
     }
   };
 
   useEffect(() => {
     fetchData();
-  }, []);
+  }, [sortOrder]);
 
   useEffect(() => {
-    if (!isLoading) {
-    localStorage.setItem("savedTodoList", JSON.stringify(todoList));
-    }
-  }, [todoList, isLoading]);
+}, [todoList]);
+
 
+  const addTodo = async () => {
+    if (!todoTitle.trim()) return;
 
-  const addTodo = (newTodo) => {
-    setTodoList((prevTodoList) => [...prevTodoList, newTodo])
+    const newTodo = {
+      fields: {
+        title: todoTitle
+      }
+    };
+
+    try {
+      const response = await fetch(API_URL, {
+        method: 'POST',
+        headers,
+        body: JSON.stringify(newTodo)
+      });
+
+      if (!response.ok) {
+        throw new Error(`Error ${response.status}`)
+      }
+
+      const data = await response.json();
+
+      if (data) {
+        setTodoList((prevTodoList) => [
+          ...prevTodoList,
+          {
+            title: data.fields.title,
+            id: data.id,
+            createdTime: data.createdTime || new Date().toISOString()
+          }
+        ]);
```

Comment:
there are two things you can do here. 
- add the new data to the old data and sort it based on the sorting order currently used, then fetch data OR
- just fetch data immediately, also disable your button when you are adding the data so that no data can be added as the request is currently on going. if you do not understand please reach out to me